src/analysis/sampling/plotSampleHist.m

```diff
@@ -119,49 +150,68 @@ function plotSampleHist(rxnNames, samples, models, nBins, perScreen, modelNames,
-        end
-        currLB = min(currLB, min(samples{i}(id,:)'));
-        currUB = max(currUB, max(samples{i}(id,:)'));
+        id = findRxnIDs(model, rxnNames{j});
+        currLB = min(currLB, min(samples{i}(id, :)'));
+        currUB = max(currUB, max(samples{i}(id, :)'));
     end
-    
-    if currUB-currLB < 8 * eps * abs(mean([currUB, currLB])) %rescale currUB, LB if they are the same # within numerical precision (8*eps)
+    % Rescale currUB, LB if they are the same # within numerical precision
+    % (8*eps)
+    if currUB - currLB < 8 * eps * abs(mean([currUB, currLB]))
         av = abs(mean([currUB, currLB]));
         currUB = currUB + 8 * eps * av;
         currLB = currLB - 8 * eps * av;
     end
-    bins = linspace(currLB, currUB, nBins);
-    
-    for i = 1:length(samples)
-        sampleSign = 1;
-        if commonModelFlag
-            id = findRxnIDs(models, rxnNameList{j});
-        else
-            id = findRxnIDs(models{i}, rxnNameList{j});
-        end
-        if isempty(id)
-            if commonModelFlag
-                id = findRxnIDs(models, [rxnNameList{j} '_r']);
-            else
-                id = findRxnIDs(models{i}, [rxnNameList{j} '_r']);
+    switch plottingApproach
+        case 'nBins'
+            clear counts;
+            
+            bins = linspace(currLB, currUB, nBins);
+            for i = 1:length(samples)
+                n = hist(1 * samples{i}(id, :), bins);
+                counts(:, i) = smooth(bins, n');
+            end
+            freq = counts ./ repmat(sum(counts), size(counts, 1), 1);
+            subplot(nX, nY, mod(j - 1, perScreen) + 1);
+            pl = plot(bins, freq, '-');
+            axis([currLB - .0001, currUB + .0001, 0, max(max(freq))]);
+
+        case 'ksdensity'
+            id = findRxnIDs(model, rxnNames{j});
+            subplot(nX, nY, mod(j - 1, perScreen) + 1);
+            hold on
+            for i = 1:length(samples)
+                eval(sprintf('sample_%s = samples{%d};', num2str(i), i));
+                eval(sprintf('ksdensity(sample_%s(id,:));', num2str(i)));
             end
-            sampleSign = -1;
+%%%%%%%%%%%%%%%%%%%%%%%%%%%% DELETE %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+eval(sprintf('plot(mean(sample_%s(id,:)), 0, ''r*'', ''LineWidth'', 2)', num2str(i)));%%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+            hold off
+    end
+    
+
+    
+    if ~isempty(FVAdata)
+        hold on
+        ylim = get(gca, 'ylim');
+        xlim = get(gca, 'xlim');
+        for i = 1:(size(FVAdata, 2) / 2)
+            %plotColor(i, :) = get(pl(round(i / 2)), 'color');  
+            patch([FVAdata(id, (i * 2) - 1) FVAdata(id, (i * 2))...
+                FVAdata(id, (i * 2)) FVAdata(id, (i * 2) - 1)],... 
+                [(((ylim(2) / 10) * i) - (ylim(2) / 10)) ...
+                (((ylim(2) / 10) * i) - (ylim(2) / 10))...
+                ((ylim(2) / 10) * i) ((ylim(2) / 10) * i)], colors{i}); %%%%%%%% borrar
```

Comment:
This can get out of bounds if `i > len(colors)`
maybe use `colors{mod(i,len(colors))+1}` ?